core/base-service/redirector.js

```diff
@@ -65,30 +66,30 @@ export default function redirector(attrs) {
         ServiceClass: this,
       })
 
-      camp.route(regex, async (queryParams, match, end, ask) => {
-        if (serverHasBeenUpSinceResourceCached(ask.req)) {
+      app.get(regex, async (req, res) => {
+        if (serverHasBeenUpSinceResourceCached(req)) {
           // Send Not Modified.
-          ask.res.statusCode = 304
-          ask.res.end()
+          res.status(304)
+          res.end()
           return
         }
 
         const metricHandle = metricHelper.startRequest()
 
-        const namedParams = namedParamsForMatch(captureNames, match, this)
+        const { namedParams, format } = paramsForReq(captureNames, req, this)
         trace.logTrace(
           'inbound',
           emojic.arrowHeadingUp,
           'Redirector',
           route.base
         )
         trace.logTrace('inbound', emojic.ticket, 'Named params', namedParams)
-        trace.logTrace('inbound', emojic.crayon, 'Query params', queryParams)
+        trace.logTrace('inbound', emojic.crayon, 'Query params', req.query)
 
         const targetPath = encodeURI(transformPath(namedParams))
         trace.logTrace('validate', emojic.dart, 'Target', targetPath)
 
-        let urlSuffix = ask.uri.search || ''
+        let urlSuffix = url.parse(req.url).search ?? '' // eslint-disable-line node/no-deprecated-api
```

Comment:
Can we do this with `let urlSuffix = new url.URL(req.url).search` instead of introducing a deprecated call here? I think that should be the same.